Essentials/src/com/earth2me/essentials/commands/Commandspawner.java

```diff
@@ -57,8 +62,31 @@ protected void run(final Server server, final User user, final String commandLab
         final Trade charge = new Trade("spawner-" + mob.name.toLowerCase(Locale.ENGLISH), ess);
         charge.isAffordableFor(user);
         try {
-            CreatureSpawner spawner = (CreatureSpawner) target.getBlock().getState();
+            Block block = target.getBlock();
+            CreatureSpawner spawner = (CreatureSpawner) block.getState();
             spawner.setSpawnedType(mob.getType());
+            if (delay > 0) {
+                if (VersionUtil.getServerBukkitVersion().isHigherThanOrEqualTo(VersionUtil.v1_12_2_R01)) {
+                    spawner.setMinSpawnDelay(0);
+                    spawner.setMaxSpawnDelay(Integer.MAX_VALUE);
+                    spawner.setMinSpawnDelay(delay);
+                    spawner.setMaxSpawnDelay(delay);
+                } else {
+                    Class<?> craftWorld = ReflUtil.getOBCClass("CraftWorld");
+                    Class<?> tileEntityMobSpawner = ReflUtil.getNMSClass("TileEntityMobSpawner");
+                    Class<?> mobSpawnerAbstract = ReflUtil.getNMSClass("MobSpawnerAbstract");
+                    Method getSpawner = tileEntityMobSpawner.getDeclaredMethod("getSpawner");
+                    Method getTileEntityAt = craftWorld.getDeclaredMethod("getTileEntityAt", int.class, int.class, int.class);
+                    Object craftTileEntity = getTileEntityAt.invoke(block.getWorld(), block.getX(), block.getY(), block.getZ());
+                    Object nmsSpawner = getSpawner.invoke(craftTileEntity);
+                    Field minSpawnDelay = ReflUtil.getFieldCached(mobSpawnerAbstract, "minSpawnDelay");
+                    Field maxSpawnDelay = ReflUtil.getFieldCached(mobSpawnerAbstract, "maxSpawnDelay");
+                    if (minSpawnDelay != null && maxSpawnDelay != null) {
+                        minSpawnDelay.setInt(nmsSpawner, delay);
+                        maxSpawnDelay.setInt(nmsSpawner, delay);
+                    }
+                }
+            }
```

Comment:
All of the Class and Field objects should be stored as a `private static final` field. Additionally, the Method objects should be stored as a `MethodHandle`. 

See https://github.com/EssentialsX/Essentials/pull/3157/files#diff-d515c857ac85b710ac67491e3dd1779aR121-R148 for an example.

Reply:
I overlooked this because I noticed `ReflUtil` already does a lot of the caching, but this is certainly a good idea. Also, didn't know about `MethodHandle`. From a quick google search, it looks like the main benefit there is it'll be a little faster to call than `Method`? I'll probably be refactoring this into it's own provider once I get some clarification from MD on how I should do that, so the NMS code won't just awkwardly be sitting inside Commandspawner.